### PR TITLE
feat(clamp) implement binary search

### PR DESCRIPTION
**Pull request by cvey**
implements #1 via binary search

### CODE REVIEW

**Comment by thiemok** [comment]
Is this redundant with #4?

**Comment by maddrag0n** [comment]
@thiemok nope #4 merged her fork/feature branch into this repo’s feature branch. this merges her fork  /feature branch into this repo’s develop branch.

File: src/tools/clamp.js [hunk 1]
```diff
@@ -27,19 +26,45 @@ const clamp = (partials: Array<string>, test: ?testFunction, {
...
+	let slicedPartials = null;
+	let half = 0;
+	let reference = partials;
+
+	if (partials.length !== 1) {
+		half = Math.floor(partials.length / 2);
+		slicedPartials = partials.slice(0, half);
+		for (let i = 0; i < slicedPartials.length; i++) {
+			words += slicedPartials[i];
+		}
```
Comment (maddrag0n) [inline src/tools/clamp.js:39]: the for loop can be written in a more elegant way:

```js
words = slicedPartials.reduce((acc, word) => acc + word, words);
```

File: src/tools/clamp.js [hunk 2]
```diff
@@ -27,19 +26,45 @@ const clamp = (partials: Array<string>, test: ?testFunction, {
...
+		reference = slicedPartials;
+	} else {
+		for (let i = 0; i < partials.length; i++) {
+			words += partials[i];
+		}
+	}
```
Comment (maddrag0n) [inline src/tools/clamp.js:45]: see above. also this is for the case when `partials.length === 1`? 😂

**Pull request closed without merging**